components/canvas_traits/webgl.rs

```diff
@@ -546,3 +547,36 @@ impl fmt::Debug for WebGLCommand {
         write!(f, "CanvasWebGLMsg::{}(..)", name)
     }
 }
+
+#[derive(Clone, Deserialize, Serialize)]
+pub struct WebGLContextLimits {
+    pub max_vertex_attribs: u32,
+    pub max_vertex_uniform_vectors: u32,
+    pub max_varying_vectors: u32,
+    pub max_vertex_texture_image_units: u32,
+    pub max_combined_texture_image_units: u32,
+    pub max_texture_image_units: u32,
+    pub max_fragment_uniform_vectors: u32,
+    pub max_tex_size: u32,
+    pub max_cube_map_tex_size: u32,
+}
+
+impl WebGLContextLimits {
+    pub fn new(gl: &gl::Gl) -> WebGLContextLimits {
```

Comment:
Thanks, great work!. Just one more thing, could you move the initialization of the struct to webgl_thread.rs (e.g.: to a separate factory method from the struct) ? I'd like to avoid adding gleam dependency to canvas_traits and doing gl code there. We are close from even removing offscreen_gl_context dependency from canvas_traits (on a separate PR I  mean). That will allow much faster compilations (e.g. from almost 10 minutos to less than 1) when developing & testing local changes in gleam & offscreen_gl because the traits are not recompiled which causes script component recompilation.

I'll merge the PR after doing that ;)